doc/dev/dialects.rst

```diff
@@ -0,0 +1,417 @@
+built-in ``func`` dialect provides everything needed to represent functions (including function
+definitions & declarations), call graphs, and inlining transformations.
+This means dialects are not monolithic program representations, but instead can be composed with
+other dialects to represent a large variety of programs. This structure is especially well-suited to
+represent heterogenous programs (like quantum algorithms).
+
+Additionally, dialects are typically organized in an *abstraction hierarchy*, with conversion rules
+describing how programs can be lowered from a high-level representation (or abstraction) to a
+low-level one. One such conversion rule might involve the decomposition of complex instructions
+into a sequence of simpler instructions, or the allocation of memory for various high-level objects.
+Alongside of optimization rules, these processes form the core part of program *compilation*.
+
+For additional information on MLIR dialects, and the type of IR objects provided by MLIR, refer to
+the following documentation pages:
+
+- `the MLIR language reference <https://mlir.llvm.org/docs/LangRef/>`_: A technical reference for
+  the MLIR language (the textual IR format), as well as its concepts and building blocks.
+- `built-in MLIR dialects <https://mlir.llvm.org/docs/Dialects/>`_: A list of dialects and
+  associated documentation for dialects developed with the MLIR project or contributed upstream.
+- `defining MLIR dialects <https://mlir.llvm.org/docs/DefiningDialects/>`_: Documentation describing
+  how to define MLIR dialects and all its components.
+- `debugging tips <https://mlir.llvm.org/getting_started/Debugging/>`_: Useful reference for any
+  kind of MLIR development, especially the debugging flags for `mlir-opt` like tools.
+
+The remainder of this document will guide you through, in concrete terms, how to add a new dialect
+to Catalyst.
+
+Catalyst dialect structure
+==========================
+
+A strong design principle in the MLIR project is the avoidance of boilerplate and redundant code,
+favoring a single source of truth to specify IR components. This is achieved via one or more
+descriptive languages based on the `TableGen <https://llvm.org/docs/TableGen/index.html>`_ format
+known from LLVM. Consequently, dialects, as well as many operations, types, attributes, and other
+IR objects, are specified in two parts:
+
+- a *declarative* part: The main way to define new IR objects via a simple language called
+  `Operation Definition Specification (ODS) <https://mlir.llvm.org/docs/DefiningDialects/Operations/>`_.
+  ODS specifications live in TableGen (``.td``) files, which will automatically generate both C++
+  header and implementation files via the build system.
+
+In Catalyst, TableGen files for dialects can be found in the public include directory, under
+`catalyst/mlir/include/\<DialectName\>/IR <https://github.com/PennyLaneAI/catalyst/tree/main/mlir/include/Quantum/IR>`_.
```

Comment:
```suggestion
  In Catalyst, TableGen files for dialects can be found in the public include directory, under
  `catalyst/mlir/include/\<DialectName\>/IR <https://github.com/PennyLaneAI/catalyst/tree/main/mlir/include/Quantum/IR>`_.
```

Reply:
This will make it render like so, which is a bit easier to read (since the 'two parts' are side by side) 
<img width="850" alt="image" src="https://github.com/user-attachments/assets/5c79adee-7424-4ad7-89e3-69b0dbc5f7e4">